### PR TITLE
Re-vendor SwarmKit to ffd009fd56a4268432b5e4ea67104a0a3a6e792c

### PR DESCRIPTION
**Pull request by aluzzardi**
Part of docker/for-linux#2

/cc @aaronlehmann @dongluochen @andrewhsu @mavenugo

### CODE REVIEW

**Comment by aaronlehmann** [comment]
LGTM

**Comment by dongluochen** [comment]
Is this regression?

```
20:23:49 ----------------------------------------------------------------------
20:23:49 FAIL: check_test.go:355: DockerSwarmSuite.TearDownTest
20:23:49 
20:23:49 check_test.go:360:
20:23:49     d.Stop(c)
20:23:49 daemon/daemon.go:392:
20:23:49     t.Fatalf("Error while stopping the daemon %s : %v", d.id, err)
20:23:49 ... Error: Error while stopping the daemon d21182562f652 : exit status 2
20:23:49 
20:23:49 
20:23:49 ----------------------------------------------------------------------
20:23:49 PANIC: docker_api_swarm_test.go:864: DockerSwarmSuite.TestAPIDuplicateNetworks
20:23:49 
20:23:49 [d21182562f652] waiting for daemon to start
20:23:49 [d21182562f652] daemon started
20:23:49 
20:23:49 docker_api_swarm_test.go:888:
20:23:49     c.Assert(err, checker.IsNil, check.Commentf(string(out)))
20:23:49 ... value *errors.errorString = &errors.errorString{s:"unexpected EOF"} ("unexpected EOF")
20:23:49 ... 
20:23:49 
20:23:49 [d21182562f652] exiting daemon
20:23:49 ... Panic: Fixture has panicked (see related PANIC)
20:23:49 
20:23:49 ----------------------------------------------------------------------
```

**Comment by dongluochen** [comment]
I run `DockerSwarmSuite.TestAPIDuplicateNetworks` on my dev machine. It succeeds on current master branch but fails on this vendor PR.

**Comment by aluzzardi** [comment]
CI failure seem legit, investigating

/cc @aaronlehmann @dongluochen @mavenugo

**Comment by aluzzardi** [comment]
@dongluochen Could you try integration-cli with `f7a002c5f1d497158c211285813c3b10d6d0f5c5`? That only contains the cherry pick

**Comment by dongluochen** [comment]
@aluzzardi When trying to vendor this cheery-pick, I get `reference is not a tree` error. f7a002c5f1d497158c211285813c3b10d6d0f5c5 might depend on other change. 

```
dchen@vm2:~/go/src/github.com/docker/docker$ vndr github.com/docker/swarmkit
2017/06/01 09:59:17 Collecting initial packages
2017/06/01 09:59:17 Download dependencies
2017/06/01 09:59:17 	Clone github.com/docker/swarmkit, revision 94645437647b2ff8dd418cd25c8d9307221e50df
2017/06/01 09:59:29 	Finished clone github.com/docker/swarmkit
2017/06/01 09:59:29 Errors on clone:
github.com/docker/swarmkit: Err: exit status 128, out: fatal: reference is not a tree: 94645437647b2ff8dd418cd25c8d9307221e50df
```

File: vendor/github.com/docker/swarmkit/api/specs.proto [hunk 1]
```diff
@@ -342,14 +342,10 @@ message NetworkSpec {
 	// "ingress" and the label "com.docker.swarm.internal": "true".
 	bool ingress = 7;
 
-	// ConfigFrom is the source of the configuration for this network.
-	oneof config_from {
-		// Network is the name of a network that provides the network
-		// specific configuration for this network, locally on the node
-		// where this network is being plumbed.
-		string network = 8;
-	}
```
Comment (aaronlehmann) [inline vendor/github.com/docker/swarmkit/api/specs.proto]: This is reverting a change that was already vendored.

**Comment by thaJeztah** [comment]
What's the status on this one? Is someone still investigating?

**Comment by thaJeztah** [comment]
closing because this is currently not mergeable, but ping @aluzzardi @aaronlehmann if we still need this

**Comment by aaronlehmann** [comment]
It's not needed anymore. Covered by #33773.

**Pull request closed without merging**